packages/asset-service/src/generateAssetData/coingecko.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  adapters,
+  avalancheChainId,
+  ChainId,
+  ethChainId,
+  fromAssetId,
+  toAssetId
+} from '@shapeshiftoss/caip'
+import { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+import lodash from 'lodash'
+
+import { avax, ethereum } from './baseAssets'
+
+type Token = {
+  chainId: number
+  address: string
+  name: string
+  symbol: string
+  decimals: number
+  logoURI: string
+}
+
+type TokenList = {
+  name: string
+  logoURI: string
+  keywords: string[]
+  timestamp: string
+  tokens: Array<Token>
+}
+export async function getAssets(
+  chainId: ChainId,
+  overrideAssets: Array<Asset> = []
+): Promise<Asset[]> {
+  const { category, explorer, explorerAddressLink, explorerTxLink } = (() => {
+    switch (chainId) {
+      case ethChainId:
+        return {
+          category: adapters.chainIdToCoingeckoAssetPlatform(chainId),
+          explorer: ethereum.explorer,
+          explorerAddressLink: ethereum.explorerAddressLink,
+          explorerTxLink: ethereum.explorerTxLink
+        }
+      case avalancheChainId:
+        return {
+          category: adapters.chainIdToCoingeckoAssetPlatform(chainId),
+          explorer: avax.explorer,
+          explorerAddressLink: avax.explorerAddressLink,
+          explorerTxLink: avax.explorerTxLink
+        }
+      default:
+        throw new Error(`no coingecko token support for chainId: ${chainId}`)
+    }
+  })()
+
+  const { data } = await axios.get<TokenList>(`https://tokens.coingecko.com/${category}/all.json`)
+
+  return data.tokens.reduce<Array<Asset>>((prev, token) => {
+    const overrideAsset = lodash.find(
+      overrideAssets,
+      (override) => fromAssetId(override.assetId).assetReference === token.address
```

Comment:
we probably want to do a `toAssetId` with the `token.address` before assuming it's the correct override asset - there may be cases where the contract address is the same across layers?

Reply:
yeah that makes sense, didn't look too closely here and just copied prev logic, but assetId compare would be much safer

Reply:
https://github.com/shapeshift/lib/pull/790/commits/17a287723d8250615d3e772289f7db26a1827ce9